_test/test_utilities_herbert/test_object_lookup.m

```diff
@@ -1,263 +1,613 @@
+            assertEqual(L.sz{1}, size(repmat(ones(Lref_sz),sz1)))
+            assertEqual(L.sz{2}, size(repmat(ones(Lref_sz),sz2)))
+            
+        end
 
-wsamp = repmat(IX_dataset_1d,size(w1_ref));
-wsamp(2) = vals2distr(xsamp(ind==2),'norm','poisson');
-wsamp(4) = vals2distr(xsamp(ind==4),'norm','poisson');
-wsamp(5) = vals2distr(xsamp(ind==5),'norm','poisson');
+        %--------------------------------------------------------------------------
+        function test_construct_nobj_2_nrep_1 (obj)
+            % Repeat multiple input object arrays using implicit repmat
+            
+            Lref = object_lookup({obj.carr1, obj.carr2});
+            [ok, mess] = check_size_and_type (Lref);
+            assertTrue(ok, mess)
+
+            Lref_nelmts = [numel(obj.carr1),numel(obj.carr2)]';
+            Lref_sz = {size(obj.carr1), size(obj.carr2)}';
+            assertEqual(Lref.narray, 2)
+            assertEqual(Lref.nelmts, Lref_nelmts)
+            assertEqual(Lref.sz, Lref_sz)
+            
+            sz1 = [3,2];    % increase size of the input object array
+            L = object_lookup({obj.carr1, obj.carr2}, '-repeat', sz1);
+            [ok, mess] = check_size_and_type (L);
+            assertTrue(ok, mess)
+            
+            assertEqual(L.object_store, Lref.object_store)
+            assertEqual(L.indx{1}, repmat(Lref.indx{1}, [prod(sz1),1]))
+            assertEqual(L.indx{2}, repmat(Lref.indx{2}, [prod(sz1),1]))
+            assertEqual(L.narray, 2)
+            assertEqual(L.nelmts, prod(sz1).*Lref_nelmts)
+            assertEqual(L.sz{1}, size(repmat(ones(Lref_sz{1}),sz1)))
+            assertEqual(L.sz{2}, size(repmat(ones(Lref_sz{2}),sz1)))
 
-if log_level>0, disp('-----------------'), end
-for i=[2,4,5]
-    [ok,mess,wdiff,chisqr] = IX_dataset_1d_same (w1_ref(i),wsamp(i),3,'rebin','chi');
-    if log_level>0
-        if ~ok
-            disp(['Dataset ',num2str(i),' BAD (chisqr = ',num2str(chisqr),') **********'])
-        else
-            disp(['Dataset ',num2str(i),' chisqr = ',num2str(chisqr)])
         end
-    end
-    if ~ok, error(['Dataset ',num2str(i),' BAD (chisqr = ',num2str(chisqr),') **********']), end
-end
+        
+        %--------------------------------------------------------------------------
+        function test_construct_nobj_2_nrep_2 (obj)
+            % Repeat multiple input object array using multiple implicit repmat
+            
+            Lref = object_lookup({obj.carr1, obj.carr2});
+            [ok, mess] = check_size_and_type (Lref);
+            assertTrue(ok, mess)
+
+            Lref_nelmts = [numel(obj.carr1),numel(obj.carr2)]';
+            Lref_sz = {size(obj.carr1), size(obj.carr2)}';
+            
+            sz1 = [3,2];
+            sz2 = [5,2,4];
+            sz12 = {sz1,sz2};    % create two increased-size input object arrays
+            L = object_lookup({obj.carr1, obj.carr2}, '-repeat', sz12);
```

Comment:
Maybe poor explanation on my part. The 'implicit' refers to what is going on inside object_lookup when called on line 273 with the '-repeat' option: it is as if you are doing repmat internally, but actually the alterations to the internal state do not involve repmat; it tinkers with indexing arrays so that the public interface returns arrays that are expanded by repmat if you explicitly want to extract them, but the internal representations are not expanded.